python/tvm/relay/op/_transform.py

```diff
@@ -123,7 +123,10 @@ def compute_scatter_add(attrs, inputs, output_type):
 @script
 def _arange_shape_func(start, stop, step):
     out = output_tensor((1,), "int64")
-    out[0] = int64(ceil_div((int64(stop[0]) - int64(start[0])), int64(step[0])))
+    if step[0] < 0:
+        out[0] = int64(ceil_div((int64(start[0]) - int64(stop[0])), int64(-step[0])))
+    else:
+        out[0] = int64(ceil_div((int64(stop[0]) - int64(start[0])), int64(step[0])))
```

Comment:
It might be a good idea to add some tests for range with negative steps elsewhere besides the onnx importer (probably `test_op_level3.py`).

Reply:
It looks like there are already tests with negative steps in test_op_level3.py, they just weren't hitting the shape_func

Reply:
Ok sounds good, thanks for confirming that!